tests/infra/proposal_generator.py

```diff
@@ -0,0 +1,334 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache 2.0 License.
+
+import argparse
+import inspect
+import json
+import os
+
+from loguru import logger as LOG
```

Comment:
Do we need a very limited `virtualenv` setup so that members can run this without pulling all the CCF infra?

Reply:
Also, I haven't run the CLI locally but I believe that the default for `loguru` is pretty verbose (i.e. prints the file name and line number on which the log function is called) which might be a little overwhelming for members running this CLI.
We could perhaps do something like in `start_network.py` instead so that the output is more concise?

Reply:
`loguru`: Great idea - I've added a simpler logging format in `__main__`.
`virtualenv`: Probably. I think this falls under future Python packaging - this is one of the bits of infra we want to expose outside, and that probably involves a Python package with narrower set of requirements.